src/app/digital/rendering/ioobjects/WireRenderer.ts

```diff
@@ -45,7 +45,13 @@ export const WireRenderer = (() => {
             ));
                 
             // @TODO move to function for getting color based on being selection/on/off
-            const color = (wire.getIsOn() ? DEFAULT_ON_COLOR : (selected ? selectedColor : wire.getColor()));
+            let color;
+            if (wire.getInput() !== undefined) {
+                color = (wire.getInput().getIsOn() ? DEFAULT_ON_COLOR : (selected ? selectedColor : wire.getColor()));
+            }
+            else {
+                color = selected ? selectedColor : wire.getColor();
+            }
```

Comment:
I think this can be condensed back down to what you had using a cool new-ish TypeScript feature called "optional chaining". Basically if you put a `?` before the `.getIsOn()` it will check if `wire.getInput()` is undefined and if it is, will break out of the statement and return `false`, but then if `wire.getInput()` is defined it will be able to call `.getIsOn()` and then work as normal ([here is some documentation on it](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining))